test/e2e/petset.go

```diff
@@ -177,6 +209,140 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
 		})
 	})
 
+	framework.KubeDescribe("Scale up/down testing [Slow] [Feature:PetSet]", func() {
+		psName := "pet"
+		labels := map[string]string{
+			"name": psName,
+		}
+		headlessSvcName := "test"
+		var controller *cache.Controller
+		var stopCh chan struct{}
+		var podTracker []podEvent
+
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("gce", "vagrant")
+
+			By("creating service " + headlessSvcName + " in namespace " + ns)
+			headlessService := createServiceSpec(headlessSvcName, "", true, labels)
+			_, err := c.Services(ns).Create(headlessService)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("configuring controller to watch for added pods")
+			labelSelector, err := unversioned.LabelSelectorAsSelector(&unversioned.LabelSelector{
+				MatchLabels: labels,
+			})
+			ExpectNoError(err)
+
+			stopCh = make(chan struct{})
+			podTracker = []podEvent{}
+			_, controller = cache.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+						options.LabelSelector = labelSelector
+						return f.Client.Pods(ns).List(options)
+					},
+					WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+						options.LabelSelector = labelSelector
+						return f.Client.Pods(ns).Watch(options)
+					},
+				},
+				&api.Pod{},
+				0,
+				cache.ResourceEventHandlerFuncs{
+					AddFunc: func(obj interface{}) {
+						podTracker = append(podTracker, podEvent{
+							pod:   obj.(*api.Pod),
+							event: add,
+						})
+					},
+					DeleteFunc: func(obj interface{}) {
+						podTracker = append(podTracker, podEvent{
+							pod:   obj.(*api.Pod),
+							event: del,
+						})
+					},
+				},
+			)
+			go controller.Run(stopCh)
+		})
+
+		AfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				dumpDebugInfo(c, ns)
+			}
+			framework.Logf("Deleting all petset in ns %v", ns)
+			deleteAllPetSets(c, ns)
+			close(stopCh)
+		})
+
+		It("Scale up/down in predictable order", func() {
```

Comment:
nit: put more info in test descriptor, see https://github.com/kubernetes/kubernetes/blob/master/docs/devel/writing-good-e2e-tests.md#some-concrete-tools-at-your-disposal